extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfReactiveConfig.java

```diff
@@ -19,6 +19,12 @@ public class CsrfReactiveConfig {
     @ConfigItem(defaultValue = "csrf-token")
     public String formFieldName;
 
+    /**
+     * Token header which will provide a CSRF token.
+     */
+    @ConfigItem
+    public Optional<String> tokenHeaderName;
```

Comment:
Why making this optional? How is it different from formFieldName?

I would be nice if we could avoid having to set a config to make this work?

Reply:
@ia3andy CSRF feature offers a double cookie submit pattern for form payloads by default,  if HTML uses form tags. If this new property is non optional, then, what does it mean for such forms, that it is ok not to provide a form parameter token or that both header and form parameter have to be provided ? Neither of these options is good for the forms, and the filter does not know how a given form has been submitted

Reply:
Right now, if the header is configured, and if it is present and the header checck passed, then the form parameter is not checked, and if it is not present then the form parameter must be present but it is not a default  behavior. If we make it a default then we tell, whatever works for the client, form param or header, will do. But the feature must stay in control and decide, via config, how it wants the token be present

Reply:
Andy, can Renarde recorder set a headet property for this feature if you prefer it be a default?

Reply:
@sberyozkin would it be a problem to assume that one OR the other (form param or header) should be present and contain a valid value to make the request valid?

Reply:
Yes, I assumed that a CSRF token would be validated either as a form element or a header, whichever one is present. It's fine to have config to disable one or the other, but by default, they should both be allowed, I think.

Reply:
you could have both as non optional `String` with default values and if the value is `false` then you disable it.

Reply:
Most clients are going to mix regular forms and ajax calls, so by default, both should work the same.

Reply:
I agree with @FroMage

Reply:
It is possible to add the header with the token inside for html form submit through htmx.